geode-wan/src/test/java/org/apache/geode/internal/cache/wan/GatewaySenderEventRemoteDispatcherJUnitTest.java

```diff
@@ -77,4 +132,160 @@ public void getConnectionShouldCreateNewConnectionWhenServerIsNull() {
     verify(dispatcher, times(1)).initializeConnection();
     verify(dispatcher, times(2)).getConnectionLifeCycleLock();
   }
+
+  @Test
+  public void enforceThreadsConnectSameReceiver_initializeParallelSenderConnection_retriesAreNotUsed() {
```

Comment:
These test names are still rather confusing in terms of what the test is doing and expecting to see. As per [the blog post on good test naming](https://osherove.com/blog/2005/4/3/naming-standards-for-unit-tests.html) I linked previously, a test name should ideally be of the form: (what work the test is doing)\_(what the conditions are)\_(what the expected result is), so for this test, a good name might be "initializeConnectionWithParalellSenderDoesNotRetryInitializeConnection".

Reply:
I read the article when you posted it the first time (interesting reading, thanks!) and tried to follow the `UnitOfWork_StateUnderTest_ExpectedBehavior` schema they propose (which btw I found it very similar to the `given-when-then` naming approach). It seems I did not understand the article in the same way as you, sorry. I used `enforceThreadsConnectSameReceiver` as `UnitOfWork` because this is what this PR is introducing.

The names you have proposed look fine for me, they really look like requirements when you read them.